src/Widget/Twig/TwigPreprocessor.php

```diff
@@ -120,7 +120,9 @@ public function preprocessEventList(array $events, $preferredLanguage, array $se
             $preprocessedEvent = $this->preprocessEvent($event, $preferredLanguage, $settings['items']);
 
             $linkType = 'query';
-            $detailUrl = $this->request->server->get('HTTP_REFERER');
+            $requestUrl = parse_url($this->request->server->get('REQUEST_URI'));
+            parse_str($requestUrl['query'], $requestQuery);
+            $detailUrl = $requestQuery['origin'];
```

Comment:
`$this->request` is a `Request` object from `symfony/http-foundation`, so you could replace this custom parsing with `$this->request->query->get('origin')`.

You could also give it an empty string as default value (second argument) so it's type-safe in case the query parameter is missing:
`$this->request->query->get('origin', '')`